### PR TITLE
chore(deps): move to aws-sdk-go-v2

### PR DESCRIPTION
**Pull request by lebauce**
## Description

The aws-sdk-go-v2 was announced as general availability
in 2021 and brings few advantages such as not storing
the endpoints list as a global variable, freeing around 2M of memory.

## Checklist
- [ ] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [ ] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/fanal/cache/s3.go [hunk 1]
```diff
@@ -17,14 +17,20 @@ import (
 
 var _ Cache = &S3Cache{}
 
+type s3API interface {
```
Comment (knqyf263) [inline pkg/fanal/cache/s3.go:20]: v2 no longer provides interfaces like v1?

**Review by knqyf263 — COMMENTED** [review]
Thanks! Looks awesome! Left a small question.

File: pkg/fanal/cache/s3.go (hunk 1 above)
Comment (lebauce) [reply pkg/fanal/cache/s3.go:20]: No, it doesn't. They just have interfaces for subparts of the API but it's incomplete (missing HeadObject and DeleteBucket) so I had to create it.

**Review by knqyf263 — APPROVED** [review]
I tested the change and it worked like a charm. Thanks!

**Pull request merged**